odxtools/diaglayer.py

```diff
@@ -636,9 +643,101 @@ def get_comparam(
 
         return cps[0]
 
-    def get_can_receive_id(self, protocol_name: Optional[str] = None) -> Optional[int]:
+    def get_can_mtu(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> Optional[int]:
+        """Return the maximum size of CAN frames that can be
+        transmitted in bytes.
+
+        For classic CAN busses, this is basically always 8. CAN-FD can
+        send up to 64 bytes per frame.
+        """
+        com_param = self.get_comparam("CP_CANFDTxMaxDataLength", protocol=protocol)
+        if com_param is None:
+            rx_id = self.get_can_receive_id(protocol=protocol)
+            if rx_id is not None:
+                # bus is CAN. We also use 8 bytes for CAN-FD if this
+                # parameter was not specified.
+                return 8
+
+            # bus is not CAN
+            return None
+
+        val = com_param.value
+        if not isinstance(val, str):
+            return None
+
+        m = re.search("TX_DL *= *([0-9]*)", val)
+        if m:
+            return int(m.group(1))
+
+        # unexpected format of parameter value
+        return 8
+
+    def use_can(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> bool:
```

Comment:
Prefixing the method with "use" means that it will do some kind of state change, and it's not a getter, which does not reflect its implementation, same for use_can_fd
```suggestion
    def get_can_used(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> bool:
```

Reply:
I don't agree with you 100%, but I see your point (to me the verb can also be a question, i.e., it does not necessarily need to be an imperative). I renamed this to [`uses_can()`](e6459950662b002641d540ffcc471db993dd3e4b). (note that I think this would better be a property, but properties cannot take arguments...)